mlflow/deployments/base.py

```diff
@@ -1,23 +1,28 @@
 """
 Here it shows the base implementation needed for plugin developers to develop deployment plugins.
-The only interfaces a deployment plugins should expose are the three functions given below
+Deployment plugin would exposes only three interfaces which is then used by mlflow to serve user
+requests.
 
-1. :py:func:`get_deploy_client`
-2. :py:func:`run_local`
-3. :py:func:`target_help`
+1. Client class subclassed from :py:class:`BaseDeploymentClient`
+2. :py:func:`run_local` for test run the deployment on local system
+3. :py:func:`target_help` for displaying the help message about the plugin
 """
 
 import abc
 
 
-def run_local(model_uri, flavor=None, config=None):
+def run_local(target, name, model_uri, flavor=None, config=None):
```

Comment:
```suggestion
def run_local(target, model_uri, flavor=None, config=None):
```

We don't use name for local deployments / allow you to manage deployments by name for local deployments, so should remove it from the API

Reply:
can't user have multiple local deployments? If we don't allow names, how'd we point to a particular deployment? For instance, in RedisAI, every deployment is a `MODELSET` command that sets the model to a redis key. While the plugin developer can create the name/key implicitly, I think it still makes sense to have that option enabled to handle multiple local deployments. What's the rationale behind not having `name` option?

Reply:
Got it, the redisai use case makes sense - for most deployment targets, deploying locally involves just running a server on some port, so users can reference locally-deployed models via their URLs (e.g. I could have one model deployed locally at http://localhost:5000 and another at http://localhost:5001).

I think it's fine to keep it in - was previously thinking all targets would ignore the `name` when deploying locally, in which case it wouldn't be a particularly useful field to keep.

Reply:
Awesome. I think it could even be useful to craft the URI like http://localhost:5000/name with the given `name` (we do something like this for a custom made REST API for one of our clients)